### PR TITLE
Fix 'observed' kwarg not doing anything on SeriesGroupBy

### LINKED ISSUE #24880: `observed` keyword for SeriesGroupBy Ignored
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd                                                                                                                                                                                                                    
df = pd.DataFrame({'a': ['x','x','y'], 'b': ['a','b','a'], 'c': [7,8,9]})                                                                                                                                                                    
df['a'] = df['a'].astype('category')
df['b'] = df['b'].astype('category')
result1 = df.groupby(['a','b']).c.agg('sum')
result2 = df.groupby(['a','b']).agg('sum')
```

#### Problem description

The calculated `result1` and `result2` DataFrames are different.
Result1:
```
a  b
x  a    7
   b    8
y  a    9
Name: c, dtype: int64
```
Result2
```
       c
a b     
x a  7.0
  b  8.0
y a  9.0
  b  NaN
```

#### Expected Output

I expect that both results have 4 rows, as the `observed` option is `False` by default.

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.6.final.0
python-bits: 64
OS: Darwin
OS-release: 17.7.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None
pandas: 0.23.4
pytest: None
pip: 18.1
setuptools: 40.6.3
Cython: None
numpy: 1.16.0
scipy: None
pyarrow: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.7.5
pytz: 2018.7
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by krsnik93**
- [x] closes #24880 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -2301,6 +2302,69 @@ def tail(self, n=5):
         mask = self._cumcount_array(ascending=False) < n
         return self._selected_obj[mask]
 
+    def _reindex_output(self, result):
```
Comment (WillAyd) [inline pandas/core/groupby/groupby.py]: Did you move this for a particular reason?

**Review by WillAyd — CHANGES_REQUESTED** [review]
I think there might be a simpler way of going about this. Does adding `_reindex_output` to `_wrap_output` in SeriesGroupBy fix it instead? I think the problem might just be that the DataFrameGroupBy wrap methods have that already but the SeriesGroupBy ones don't, so that could help

File: pandas/core/groupby/groupby.py (hunk 1 above)
Comment (WillAyd) [reply pandas/core/groupby/groupby.py]: Nevermind I see the reasoning here since this wasn't available to SeriesGroupBy before

**Comment by jreback** [comment]
@krsnik93 this PR might be ok

**Pull request closed without merging**